app/full_text_search.py

```diff
@@ -213,6 +242,12 @@ def create_romanized_name_fields(romanize_method, **kwargs):
         if romanized_name:
             fields.extend(create_fields_for_rank(field, romanized_name))
             romanized_names_list.append(romanized_name)
+        if script_variant.has_kanji(kwargs[field]):
```

Comment:
You may not limit to kanji. You may always add this _non_romanized field and rank exact matches higher than others. What do you think?